src/components/add-account-dialog/add-account-dialog.js

```diff
@@ -69,7 +69,10 @@ export class AddAccountDialogComponent {
       name: this.accountNameInput.value,
       balance: parseInt(this.balanceNameInput.value),
       type: this.accountType.innerText,
-      currency: this.accountCurrency.innerText
+      currency: this.accountCurrency.innerText,
+      id: Math.random()
```

Comment:
As far as I know you're updating account in your service by name -- so either use id's everywhere or do not use them at all

Reply:
I found it here https://gist.github.com/gordonbrander/2230317

Reply:
ok